sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -0,0 +1,1050 @@
+preemptively changed the state of the queue.
+
+=== Error Handling
+
+Errors are reported through exceptions, as usual in the SYCL API. For new APIs,
+submitting a graph for execution can generate unspecified asynchronous errors,
+while `command_graph::finalize()` may throw unspecified synchronous exceptions.
+Synchronous exception errors codes are defined for both
+`queue::begin_recording()` and `command_graph::update()`.
+
+When a queue is in recording mode asynchronous exceptions will not be
+generated, as no device execution is occuring. Synchronous errors specified as
+being thrown in the default queue executing state, will still be thrown when a
+queue is in the recording state.
+
+The `queue::begin_recording` and `queue::end_recording` entry-points return a
+`bool` value informing the user whether a state change occurred. False is
+returned rather than throwing an exception when state isn't changed. This design
+is because the queue is already in the state the user desires, so if the
+function threw an exception in this case, the application would likely swallow
+it and then proceed.
+
+While a queue is in the recording state, methods performed on that queue which
+are not command submissions behave as normal. This includes waits, throws, and
+queries on the queue. These are all ignored by the graph system, as opposed to
+throwing an exception when in queue recording mode. This is because otherwise
+there would be no thread safe way for a user to check they could call these
+functions without throwing, as a query about the state of the queue may be
+immediately stale.
+
+=== Storage Lifetimes
+
+The lifetime of any buffer recorded as part of a submission
+to a command graph will be extended in keeping with the common reference
+semantics and buffer synchronization rules in the SYCL specification. It will be
+extended either for the lifetime of the graph (including both modifiable graphs
+and the executable graphs created from them) or until the buffer is no longer
+required by the graph (such as after being replaced through executable graph update).
+
+=== Host Tasks
+
+:host-task: https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:interfaces.hosttasks
+
+A {host-task}[host task] is a native C++ callable, scheduled according to SYCL
+dependency rules. It is valid to record a host task as part of graph, though it
+may lead to sub-optimal graph performance because a host task node may prevent
+the SYCL runtime from submitting the whole graph to the device at once.
+
+Host tasks can be updated as part of <<executable-graph-update, executable graph update>>
+by replacing the whole node with the new callable.
+
+== Examples
+
+[NOTE]
+====
+The examples below demonstrate intended usage of the extension, but may not be
+compatible with the proof-of-concept implementation, as the proof-of-concept
+implementation is currently under development.
+====
+
+=== Dot Product
+
+[source,c++]
+----
+...
+
```

Comment:
Might be worth adding a comment that this example is for demonstration prurposes only and leaves out how the input data is set.